src/alloc-aligned.c

```diff
@@ -14,10 +14,13 @@ terms of the MIT license. A copy of the license can be found in the file
 // Aligned Allocation
 // ------------------------------------------------------
 
-static void* mi_heap_malloc_zero_aligned_at(mi_heap_t* const heap, const size_t size, const size_t alignment, const size_t offset, const bool zero) mi_attr_noexcept {
+static void* mi_heap_malloc_zero_aligned_at(mi_heap_t* const heap, const size_t size, size_t alignment, const size_t offset, const bool zero) mi_attr_noexcept {
   // note: we don't require `size > offset`, we just guarantee that
   // the address at offset is aligned regardless of the allocated size.
-  mi_assert(alignment > 0 && alignment % sizeof(void*) == 0);
+
+  // if the passed alignment is too small, correct it upwards to the minimal internal required alignment
+  // fixes issues in user code calling aligned variants for types with small alignment requirements below pointer alignment
+  alignment = alignment < sizeof(void*) ? sizeof(void*) : alignment;
```

Comment:
I think this should be below the if()s below which check that we were fed with a non-0 power-of-two first, before alignment is changed.  Those requirements are sane.